ansible_collections/arista/avd/plugins/filter/convert_dicts.py

```diff
@@ -43,38 +42,44 @@ def convert_dicts(dictionary, primary_key="name", secondary_key="items"):
     Parameters
     ----------
     dictionary : any
-        Nested Dictionary to convert - returned untouched if not a nested dictionary
+        Nested Dictionary to convert - returned untouched if not a nested dictionary and list
     primary_key : str, optional
         Name of primary key used when inserting outer dictionary keys into items.
     secondary_key : str, optional
-        Name of secondary key used when inserting dictionary values which are list of dictionaries into items.
+        Name of secondary key used when inserting dictionary values which are list into items.
 
     Returns
     -------
     any
-        Returns list of dictionaries or input variable untouched if not a nested dictionary/list of dictionaries
+        Returns list of dictionaries or input variable untouched if not a nested dictionary/list.
     """
-    if not isinstance(dictionary, dict) or os.environ.get('AVD_DISABLE_CONVERT_DICTS'):
+    if not isinstance(dictionary, dict) and not isinstance(dictionary, list) or os.environ.get('AVD_DISABLE_CONVERT_DICTS'):
         # Not a dictionary, return the original
         return dictionary
+    elif isinstance(dictionary, list):
+        output = []
+        for element in dictionary:
```

Comment:
@ClausHolbechArista , is this how it should be? (need a confirmation).
currently it works like this.

Reply:
@gmuloc correct, that is what we have today. We do not enforce the primary key to be in the dict, but we convert it to _be_ a dict.

Reply:
@ClausHolbechArista but so the expectation from the filter is that when we pass dicts or list they are all "containing" the same type of elements? If so should probably be added because in my example a list of `[<string>, <dict>]` becomes `[<dict>, <dict>]` but the second dict does not have `name` as a key (which probably the person running the filter could expect)

Reply:
looks like this is blocking the two approvals so will resolve here as it is the expected behavior for now.

Reply:
We can try to improve the docs later.